.github/workflows/check.yml

```diff
@@ -140,3 +140,24 @@ jobs:
       - name: cargo-semver-checks
         if: matrix.rust-toolchain == 'stable'
         run: cargo semver-checks
+
+  check-freebsd:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Get "Install Rust" action from neqo
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          repository: mozilla/neqo
+          sparse-checkout: |
+            .github/actions/rust
+          path: neqo
+
+      - name: Install Rust
+        uses: ./neqo/.github/actions/rust
+
+      - run: rustup target add x86_64-unknown-freebsd
+
+      - run: cargo check --target x86_64-unknown-freebsd
```

Comment:
Is this actually useful? Shouldn't we rather add a proper `vmaction`-based FreeBSD workflow like @miguelcsx has? Because this will only work for the compile of the placeholder function we have now and not an actual FreeBSD code path that relies on its system calls,

Reply:
Agreed, long term we should run `vmaction`-based FreeBSD. I assumed that was the plan to be added with FreeBSD support itself.

In the meantime I just wanted some (quick) way to ensure we don't regress until then.

Reply:
What do you think about having both `check-freebsd` and `test-freebsd` jobs, one for quick operation checks as @mxinden  says, and the other for long-term support, I could make sure they are in the same job, or separate them as mentioned in #19?

Reply:
That would work for me as well. I'd still like to run them as separate (matrix) workflows, so CI finishes faster overall.